src/main/InternalProcess.js

```diff
@@ -0,0 +1,150 @@
+import { fork } from "child_process";
+import logger from "~/logger";
+
+class InternalProcess {
+  constructor(timeout) {
+    this.process = null;
+
+    this.timeout = timeout;
+    this.active = false;
+
+    this.onStartCallback = null;
+    this.onMessageCallback = null;
+    this.onExitCallback = null;
+
+    this.messageQueue = [];
+    this.sending = false;
+  }
+
+  onMessage(callback) {
+    this.onMessageCallback = callback;
+  }
+
+  onExit(callback) {
+    this.onExitCallback = callback;
+  }
+
+  run() {
+    this.sending = true;
+    while (this.messageQueue.length && this.active) {
+      const message = this.messageQueue.shift();
+      this.process.send(message);
+    }
+    this.sending = false;
```

Comment:
I don't understand why sending is necessary. this code is synchronous and there is no way, outside this function that `sending` can be true.